src/pages/ExternalSupplierInvoicePage.js

```diff
@@ -0,0 +1,441 @@
+    const infoColumns = [
+      [
+        {
+          title: `${pageInfoStrings.entry_date}:`,
+          info: formatDate(transaction.entryDate),
+        },
+        {
+          title: `${pageInfoStrings.confirm_date}:`,
+          info: formatDate(transaction.confirmDate),
+        },
+        {
+          title: `${pageInfoStrings.total_price}:`,
+          info: this.state.totalPrice,
+        },
+      ],
+      [
+        {
+          title: `${pageInfoStrings.supplier}:`,
+          info: this.props.transaction.otherParty && this.props.transaction.otherParty.name,
+        },
+        {
+          title: `${pageInfoStrings.their_ref}:`,
+          info: transaction.theirRef,
+          onPress: this.openTheirRefEditor,
+          editableType: 'text',
+        },
+        {
+          title: `${pageInfoStrings.comment}:`,
+          info: transaction.comment,
+          onPress: this.openCommentEditor,
+          editableType: 'text',
+        },
+      ],
+    ];
+    return (
+      <PageInfo columns={infoColumns} isEditingDisabled={this.props.transaction.isFinalised} />
+    );
+  }
+
+  renderCell(key, transactionBatch) {
+    const isEditable = !this.props.transaction.isFinalised;
+    const type = isEditable ? 'editable' : 'text';
+    switch (key) {
+      default:
+        return transactionBatch[key];
+      case 'packSize': {
+        const renderedCell = {
+          type: type,
+          cellContents: String(transactionBatch.packSize),
+        };
+        return renderedCell;
+      }
+      case 'numberOfPacks': {
+        const renderedCell = {
+          type: type,
+          cellContents: String(transactionBatch.numberOfPacks),
+        };
+        return renderedCell;
+      }
+      case 'batch': {
+        const renderedCell = {
+          type: type,
+          cellContents: transactionBatch.batch,
+          keyboardType: 'default',
+        };
+        return renderedCell;
+      }
+      case 'costPrice': {
+        const renderedCell = {
+          type: type,
+          cellContents: String(transactionBatch.costPrice),
+        };
+        return renderedCell;
+      }
+      case 'expiryDate': {
+        const expiryDate = formatExpiryDate(transactionBatch.expiryDate);
+        const renderedCell = {
+          type: type,
+          cellContents: expiryDate || 'month/year',
+        };
+        return renderedCell;
+      }
+      case 'remove':
+        return {
+          type: 'checkable',
+          icon: 'md-remove-circle',
+          isDisabled: this.props.transaction.isFinalised,
+        };
+      // expiryDate
+    }
+  }
+
+  addNewSIitem(item) {
+    const { database, transaction } = this.props;
+    database.write(() => {
+      const transactionItem = createRecord(database, 'TransactionItem', transaction, item);
+      createRecord(
+        database,
+        'TransactionBatch',
+        transactionItem,
+        createRecord(database, 'ItemBatch', item, '')
```

Comment:
from what I've seen in mSupply desktop it is possible to have multiple batch names (this is where 'merge' comes in).. I do agree a better solution will be to create itemBatch on finalise as per my previous comment.. I don't think it will solve the multiple batch problem though.. the juice of stocktake batch creating is in Stocktake.adjustInventory .. ill check it out see what is common